pyfuncol/list.py

```diff
@@ -158,6 +158,107 @@ def index_of(self: List[A], elem: A) -> int:
             return i
     return -1
 
+def fold_left(self: List[A], z: B, op: Callable[[B, A], B]) -> B:
+    """
+    Applies a binary operator to a start value and all elements of this sequence, going left to right.
+
+    Args:
+        z: start value
+        op: binary operation
+
+    Returns:
+        the result of inserting op between consecutive elements of this sequence, going left to right with the start value z on the left:
+        op(...op(z, x_1), x_2, ..., x_n)
+        where x1, ..., xn are the elements of this sequence. Returns z if this sequence is empty.
+    """
+    acc = z
+    for x in self:
+        acc = op(acc, x)
+    return acc
+
+def fold_right(self: List[A], z: B, op: Callable[[A, B], B]) -> B:
+    """
+    Applies a binary operator to all elements of this list and a start value, going right to left.
+
+    Args:
+        z: start value
+        op: binary operation
+
+    Returns:
+        the result of inserting op between consecutive elements of this list, going right to left with the start value z on the right:
+        op(x_1, op(x_2, ... op(x_n, z)...))
+        where x1, ..., xn are the elements of this list. Returns z if this list is empty.
+    """
+    
+    acc = z
+    for x in reversed(self):
+        acc = op(x, acc)
+    return acc
+
+def forall(self: List[A], p: Callable[[A], bool]) -> bool:
+    """
+    Tests whether a predicate holds for all elements of this list.
+
+    Args:
+        p: the predicate used to test elements.
+        returns: true if this list is empty or the given predicate p holds for all elements of this list, otherwise false.
+    """
+    for x in self:
+        if not p(x):
+            return False
+    
+    return True
+
+def head(self: List[A]) -> A:
+    """
+    Selects the first element of this iterable collection.
+
+    Note: might return different results for different runs, unless the underlying collection type is ordered.
+
+    Exceptions thrown:
```

Comment:
You should use instead `Raises:` as it is understood by the docs when they autogenerate the API reference

Reply:
👍